cupyx/scipy/sparse/compressed.py

```diff
@@ -433,6 +436,85 @@ def _with_data(self, data, copy=True):
                 shape=self.shape,
                 dtype=data.dtype)
 
+    def prune(self):
+        """Remove empty space after all non-zero elements.
+        """
+        major_dim = self._swap(*self.shape)[0]
+
+        if self.indptr.size != major_dim + 1:
+            raise ValueError('index pointer has invalid length')
+
+        if self.indices.size < self.indptr[-1]:
+            raise ValueError('indices array has fewer than nnz elements')
+
+        # NOTE: With the current getnnz implementation, this would
+        # never fail because getnnz returns self.data.size. Using
+        # indptr[-1] instead.
+        if self.data.size < self.indptr[-1]:
+            raise ValueError('data array has fewer than nnz elements')
+
+        self.indices = util.prune_array(self.indices[:self.indptr[-1]])
+        self.data = util.prune_array(self.data[:self.indptr[-1]])
+
+    def check_format(self, full_check=True):
+        """Check whether the matrix format is valid
+
+        Args:
+            full_check (bool, optional): If `True`, rigorous check,
+                O(N) operations. Otherwise basic check, O(1) operations
+                (default True).
+        """
+        # use _swap to determine proper bounds
+        major_name, minor_name = self._swap('row', 'column')
+        major_dim, minor_dim = self._swap(*self.shape)
+
+        # index arrays should have integer data types
+        if self.indptr.dtype.kind != 'i':
+            warn("indptr array has non-integer dtype ({})"
+                 "".format(self.indptr.dtype.name), stacklevel=3)
+        if self.indices.dtype.kind != 'i':
+            warn("indices array has non-integer dtype ({})"
+                 "".format(self.indices.dtype.name), stacklevel=3)
+
+        idx_dtype = sputils.get_index_dtype((self.indptr, self.indices))
+        self.indptr = cupy.asarray(self.indptr, dtype=idx_dtype)
```

Comment:
`cupy.asarray` can convert all array-like inputs to `cupy.ndarray`. Please use `astype` if the input is guaranteed to be `cupy.ndarray`.
```suggestion
        self.indptr = self.indptr.astype(dtype=idx_dtype)
```